### PR TITLE
Feat/us22-24 admin invoices and merchants

### PR DESCRIPTION
**Pull request by AuroraTuring**

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -6,4 +6,8 @@ class Invoice < ApplicationRecord
   has_many :merchants, through: :items
 
   enum :status, ["cancelled", "in progress", "completed"]
-end
\ No newline at end of file
+
+  def self.unshipped_invoices
+    Invoice.joins(:invoice_items).where("invoice_items.status != 2").group("invoices.id").order(:created_at)
+  end
```
Comment (NeilTheSeal) [inline app/models/invoice.rb:12]: This finds invoice items that are status = 1 or status = 0, and the `group("invoices.id")` basically just takes out the duplicate invoices

File: app/models/invoice.rb [hunk 2]
```diff
@@ -6,4 +6,8 @@ class Invoice < ApplicationRecord
   has_many :merchants, through: :items
 
   enum :status, ["cancelled", "in progress", "completed"]
-end
\ No newline at end of file
+
+  def self.unshipped_invoices
```
Comment (LuisAparicio14) [inline app/models/invoice.rb:10]: This method is similar to the method that we created for user story 4 and 5.

File: app/views/admin/index/index.html.erb [hunk 3]
```diff
@@ -4,6 +4,16 @@
   
   <div class="content-column">
     <div class="content-header">Incomplete Invoices</div>
+    <ol class="content-ordered-list">
+      <% @unshipped_invoices.each do |invoice| %>
+        <li class="ordered-list-item">
+          <%= link_to "Invoice ##{invoice.id}", "/admin/invoices/#{invoice.id}" %>
+          <div class="date"> 
+            <%= invoice.created_at.strftime("%A, %B %d, %Y") %>
```
Comment (NeilTheSeal) [inline app/views/admin/index/index.html.erb:12]: This formats a date to be something like `Wednesday, January 01, 2021`

File: spec/features/admin/index/index_spec.rb [hunk 4]
```diff
@@ -3,13 +3,29 @@
...
-    @customer2_invoice_list = create_list(:invoice, 3, customer: @customers[1])
-    @customer3_invoice_list = create_list(:invoice, 5, customer: @customers[2])
+    @item_list1 = create_list(:item, 3, merchant: @merchant, unit_price: 1000)
+    @item_list2 = create_list(:item, 3, merchant: @merchant, unit_price: 1000)
+    @item_list3 = create_list(:item, 3, merchant: @merchant, unit_price: 1000)
+
+
+    @customer1_invoice_list = create_list(:invoice, 1, customer: @customers[0], created_at: "2020-01-01 00:00:00")
+    @customer2_invoice_list = create_list(:invoice, 3, customer: @customers[1], created_at: "2022-01-01 00:00:00")
+    @customer3_invoice_list = create_list(:invoice, 5, customer: @customers[2], created_at: "2021-01-01 00:00:00")
```
Comment (NeilTheSeal) [inline spec/features/admin/index/index_spec.rb:15]: You can manually set the `created_at` column for any factorybot object using the syntax above

File: app/views/admin/index/index.html.erb (hunk 3 above)
Comment (LuisAparicio14) [inline app/views/admin/index/index.html.erb:12]: I like the formatting of the created_at with the specific date.

File: app/models/invoice.rb [hunk 5]
```diff
@@ -6,4 +6,8 @@ class Invoice < ApplicationRecord
...
+    Invoice.joins(:invoice_items).where("invoice_items.status != 2").group("invoices.id").order(:created_at)
```
Comment (grantdavis303) [inline app/models/invoice.rb:11]: Nice! I like how y'all chose invoice items != to 2 as opposed to either equaling 0 or 1, makes it a lot cleaner

File: spec/features/admin/index/index_spec.rb [hunk 6]
```diff
@@ -47,4 +63,27 @@
...
+  end
+
+  it "has each invoice id which links to that invoice's admin show page" do
+    visit "/admin"
+
+    click_link("Invoice ##{@customer2_invoice_list[0].id}")
+    expect(page).to have_current_path("/admin/invoices/#{@customer2_invoice_list[0].id}")
+  end
+
+  it "Next to each invoice id I see the date that the invoice was created" do
```
Comment (NeilTheSeal) [inline spec/features/admin/index/index_spec.rb:84]: This should also be ordered by created_at in ascending order

File: spec/features/admin/index/index_spec.rb [hunk 7]
```diff
@@ -3,13 +3,29 @@
 RSpec.describe "Admin dashboard" do
   before(:each) do
```
Comment (LuisAparicio14) [inline spec/features/admin/index/index_spec.rb:4]: This before each looks better than ours

**Review by LuisAparicio14 — APPROVED** [review]

File: spec/features/admin/index/index_spec.rb [hunk 8]
```diff
@@ -47,4 +63,27 @@
...
+    visit "/admin"
+
+    click_link("Invoice ##{@customer2_invoice_list[0].id}")
+    expect(page).to have_current_path("/admin/invoices/#{@customer2_invoice_list[0].id}")
+  end
+
+  it "Next to each invoice id I see the date that the invoice was created" do
+    visit "/admin"
+
+    expect("Friday, January 01, 2021").to appear_before("Saturday, January 01, 2022")
```
Comment (grantdavis303) [inline spec/features/admin/index/index_spec.rb:87]: Solid feature tests. I'm still learning about factory bot and faker but the invoice lists are starting to make sense!

**Review by grantdavis303 — APPROVED** [review]

**Pull request merged**